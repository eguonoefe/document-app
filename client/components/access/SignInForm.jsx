import React, { useState, Fragment } from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

/**
 * SignIn component for logging in a user
 * @param {Object} props { logInUser }
 * @returns {String} The HTML markup for the SignInForm
 */
const SignInForm = ({ logInUser }) => {

  const [values, setValues] = useState({email: '', password: ''});

  const onChange = e => {
    const {name, value} = e.target;
    setValues({ ...values, [name]: value });
  }

  const submit = () => logInUser(values)

  return (
    <Fragment>
      <h5 className="center"> Log Into Doc-ms</h5>
      <div className="input-field">
        <input
          className="validate form-design"
          type="email"
          name="email"
          value={values.email}
          onChange={onChange}
          placeholder="Email"
          required
        />
      </div>
      <div className="input-field">
        <input
          className="validate form-design"
          type="password"
          name="password"
          value={values.password}
          onChange={onChange}
          placeholder="password"
          required
        />
      </div>
      <div className="input-field center">
        <button
          id="submit-signin"
          className="waves-effect btn button-design"
          onClick={submit}
        >
          Sign In
        </button>
      </div>
      <p className="center">
        Don&#39;t Have an account? <Link to="signup">Create an Account</Link>
      </p>
    </Fragment>
  );
};

SignInForm.propTypes = {
  logInUser: PropTypes.func.isRequired
};

export default SignInForm;
