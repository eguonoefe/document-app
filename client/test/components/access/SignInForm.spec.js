/* global expect jest test */
import React from 'react';
import { shallow, configure } from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';

import SignInForm from '../../../components/access/SignInForm';

configure({ adapter: new Adapter() });

describe('SignInForm Component', () => {
  const logInUser = jest.fn();
  const component = shallow(<SignInForm logInUser={logInUser} />);
  test('should match the SignInForm snapshot', () => {
    const tree = toJson(component);
    expect(component.find('input').length).toEqual(2);
    expect(tree).toMatchSnapshot();
  });
});
