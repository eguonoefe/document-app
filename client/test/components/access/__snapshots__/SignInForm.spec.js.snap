// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignInForm Component should match the SignInForm snapshot 1`] = `
<Fragment>
  <h5
    className="center"
  >
     Log Into Doc-ms
  </h5>
  <div
    className="input-field"
  >
    <input
      className="validate form-design"
      name="email"
      onChange={[Function]}
      placeholder="Email"
      required={true}
      type="email"
      value=""
    />
  </div>
  <div
    className="input-field"
  >
    <input
      className="validate form-design"
      name="password"
      onChange={[Function]}
      placeholder="password"
      required={true}
      type="password"
      value=""
    />
  </div>
  <div
    className="input-field center"
  >
    <button
      className="waves-effect btn button-design"
      id="submit-signin"
      onClick={[Function]}
    >
      Sign In
    </button>
  </div>
  <p
    className="center"
  >
    Don't Have an account? 
    <Link
      to="signup"
    >
      Create an Account
    </Link>
  </p>
</Fragment>
`;
